File: 3471-find-the-largest-almost-missing-integer/Solution.py
class Solution:
    def largestInteger(self, nums, k):
        count = {}

        for i in range(len(nums) - k + 1):
            seen = set(nums[i:i + k])

            for value in seen:
                count[value] = count.get(value, 0) + 1

        answer = -1

        for value, frequency in count.items():
            if frequency == 1:
                answer = max(answer, value)

        return answer

File: 3471-find-the-largest-almost-missing-integer/Solution.java
import java.util.*;

class Solution {
    public int largestInteger(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<>();

        for (int i = 0; i <= nums.length - k; i++) {
            Set<Integer> seen = new HashSet<>();

            for (int j = i; j < i + k; j++) {
                seen.add(nums[j]);
            }

            for (int value : seen) {
                count.put(value, count.getOrDefault(value, 0) + 1);
            }
        }

        int answer = -1;

        for (Map.Entry<Integer, Integer> entry : count.entrySet()) {
            if (entry.getValue() == 1) {
                answer = Math.max(answer, entry.getKey());
            }
        }

        return answer;
    }
}
